ssl/statem/statem_srvr.c

```diff
@@ -3924,9 +3925,13 @@ int tls_construct_new_session_ticket(SSL *s, WPACKET *pkt)
                 SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_MALLOC_FAILURE);
                 goto err;
             }
+
+            /* s->session->ext is CBB extensions in bssl */
```

Comment:
Er, it's not. `CBB` is our thing for assembling a message to serialize somewhere, while `s->session->ext` is some data stored in the `SSL_SESSION`. The point of GREASE is to actually send it over the wire. You also probably don't want to store it in the `SSL_SESSION`, which gets carried over across resumptions.

Reply:
Thanks David I'll look into this in more detail!

Reply:
The session thing you explain makes complete sense, it's something I wasn't confident on when I wrote this. Thanks for noticing my assumption in the comment.